doc/classes/EditorImportPlugin.xml

```diff
@@ -241,7 +241,80 @@
+				    var source_file_folder: String = source_file.rsplit("/", true, 1)[0]
+				    var textures_folder: String = "%s/textures" % source_file_folder
+				    DirAccess.make_dir_recursive_absolute(textures_folder)
+
+				    # Simulate image export by copying the source file to the textures folder.
+				    var export_path: String = "%s/icon.png" % textures_folder
+				    var export_image_file: FileAccess = FileAccess.open(export_path, FileAccess.WRITE)
+				    export_image_file.store_buffer(FileAccess.get_file_as_bytes("res://game/my_icon.png"))
+				    export_image_file.close()
+
+				    # Refresh view of exported file.
+				    var editor_file_system: EditorFileSystem = _editor.get_resource_filesystem()
+				    editor_file_system.update_file(textures_folder)
+				    editor_file_system.update_file(export_path)
+
+				    # Append exported file as external resource.
+				    var err: Error = self.append_import_external_resource(export_path)
+				    if err != OK:
+				        return err
+
+				    # Load exported image as a Texture2D.
+				    var export_image_tex: Texture2D = ResourceLoader.load(export_path, "Texture2D", ResourceLoader.CACHE_MODE_REPLACE)
+				    if export_image_tex == null:
+				        return ERR_SCRIPT_FAILED
+
+				    ...
+				[/gdscript]
+				[csharp]
+				public partial class MyEditorImportPlugin : EditorImportPlugin
+				{
+				    public override Error _Import(string sourceFile, string savePath, Godot.Collections.Dictionary options, Godot.Collections.Array&lt;string&gt; platformVariants, Godot.Collections.Array&lt;string&gt; genFiles)
+				    {
+				        // Make textures folder if required.
+				        string sourceFileFolder = sourceFile.Substring(0, sourceFile.LastIndexOf('/'));
+				        string texturesFolder = $"{sourceFileFolder}/Textures";
+				        DirAccess.MakeDirRecursiveAbsolute(texturesFolder);
+
+				        // Simulate image export by copying the source file to the textures folder.
+				        string exportPath = $"{texturesFolder}/Icon.png";
+				        using FileAccess exportImageFile = FileAccess.Open(exportPath, FileAccess.ModeFlags.Write);
+				        exportImageFile.StoreBuffer(FileAccess.GetFileAsBytes("res://Game/MyIcon.png"));
+				        exportImageFile.Close();
+
+				        // Ensure file visibility for ResourceLoader later.
+				        EditorFileSystem editorFileSystem = EditorInterface.Singleton.GetResourceFilesystem();
+				        editorFileSystem.UpdateFile(texturesFolder);
+				        editorFileSystem.UpdateFile(exportPath);
+
+				        // Append exported file as external resource.
+				        Error err = AppendImportExternalResource(exportPath);
+				        if (err != Error.Ok)
+				        {
+				            return err;
+				        }
+
+				        // Load exported image as a Texture2D.
+				        Texture2D exportImageTex = ResourceLoader.Load&lt;Texture2D&gt;(exportPath, nameof(Texture2D), ResourceLoader.CacheMode.Replace);
```

Comment:
The original PR was using a string literal instead of `nameof`. I think it's fine to use `nameof` but keep in mind C# class names don't always match the name of the class in ClassDB (i.e.: `AudioStreamWav` instead of `AudioStreamWAV`).